Assets/Scripts/System/ObjectPooler.cs

```diff
@@ -0,0 +1,84 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ObjectPooler : MonoBehaviour
+{
+
+    //We do a little pooling
+    #region Explanation
+    //Example of spawning something from a pool that has been already assigned in the inspector
+    // GameObject objectToSpawn = ObjectPooler.Instance.SpawnFromPool("pool_name", position, rotation);
+    //When it's time to discard an object in a pool, instead of doing a Destroy, SetActive(false) instead
+    //When you spawn an object from a pool, if the gameobject has a default state, you will need to reset them again OnEnable
+    //We can possibly have a reset interface for this
+    //This is only helpful for objects we constantly spawn over and over, like a dust effect on a character stomping around, or bullets from a gun
+    //Can also be used to limit the amount of objects of the same type in a scene, I.E we don't want more than 10 health pickups on the floor
+    #endregion
+
+    [System.Serializable]
+    public class Pool
+    {
+        //Our available pools, set in the inspector
+        [Tooltip("Name of this pool")]
+        public string tag;
+        [Tooltip("Object occupying this pool")]
+        public GameObject prefab;
+        [Tooltip("Amount of objects this pool will be populated with on scene load, if you spawn too few, you may see objects disappear and reappear at the new spawn location")]
+        public int size;
+    }
+
+    #region Variables_Instance
+    public List<Pool> pools;
+    public Dictionary<string, Queue<GameObject>> poolDictionary;
```

Comment:
So your using a dictionary to be a reference pointer? 
Will your array of spawned objects ever need a system to search through the list? 
Will your list be used in the unityEditor for testing? Could you change this to be a hashTable only? 1 variable instead of 2 taking memory?